azure/services/managedclusters/spec.go

```diff
@@ -395,6 +398,21 @@ func (s *ManagedClusterSpec) Parameters(ctx context.Context, existing interface{
 
 	managedCluster.AutoScalerProfile = buildAutoScalerProfile(s.AutoScalerProfile)
 
+	if s.IdentityProfile != nil {
+		managedCluster.Identity = &containerservice.ManagedClusterIdentity{
+			Type: containerservice.ResourceIdentityType(s.IdentityProfile.Type),
+		}
+		if s.IdentityProfile.Type == infrav1.ManagedControlPlaneIdentityTypeUserAssigned {
+			userAssignedIDValues := &containerservice.ManagedClusterIdentityUserAssignedIdentitiesValue{
+				PrincipalID: s.IdentityProfile.UserAssignedIdentityPrincipalID,
+				ClientID:    s.IdentityProfile.UserAssignedIdentityClientID,
+			}
+			managedCluster.Identity.UserAssignedIdentities = map[string]*containerservice.ManagedClusterIdentityUserAssignedIdentitiesValue{
```

Comment:
@alexeldeib thanks for the review (and sorry for the delayed response)

Are you suggesting that these are the wrong inputs to pass into the AKS API?